core/src/main/java/org/elasticsearch/index/query/IndicesQueryBuilder.java

```diff
@@ -31,48 +39,67 @@
 
     public static final String NAME = "indices";
 
-    private final QueryBuilder queryBuilder;
+    /** The default query to be executed on non matching indices */
+    public static final QueryBuilder DEFAULT_NO_MATCH_QUERY = QueryBuilders.matchAllQuery();
```

Comment:
I think this public constant is a bit scary, as it's an object and can be modified (think queryName or boost). I think in this case I would not add a constant for the default value, create a new query all the time.